src/lib/peer_trust_lib/peer_trust.mli

```diff
@@ -0,0 +1,73 @@
+(** Module handling peer "trust" scores. Trust scores increase when peers do
+    good things and decrease when they do bad things. If a peer does enough bad
+    things it gets banned. This module tracks, stores and updates trust, and
+    notifies the rest of the system when a peer is banned. It does *not* specify
+    specific things that cause trust to change, or how much, or actually do
+    anything when a peer should be banned. That's the responsibility of the
+    caller, which is coda_base. *)
+
+open Core
+
+module type Action = sig
+  (* TODO add a show interface for logging. *)
+  type t
+
+  val to_trust_increment : t -> float
+end
+
+val insta_ban : float
+(** Trust increment to instantly ban the peer. *)
+
+val max_rate : float -> float
+(** Trust increment that sets a maximum rate of doing a bad thing (presuming the
+    peer does no good things) in seconds/action. *)
+
+(* TODO consider deduplicating somehow, maybe with an intf.ml, or by getting rid
+   of this definition entirely. *)
+
+(** An instantiated peer trust module *)
+module type S = sig
+  type t
+
+  (** The data identifying a peer. *)
+  type peer
+
+  (** Some action a peer took that affects trust. *)
+  type action
+
+  val create : db_dir:string -> t
+  (** Set up the trust system. *)
+
+  (* TODO pipe for ban events *)
+
+  val record : t -> peer -> action -> unit
+  (** Record an action a peer took. This may result in a ban event being
+      emitted *)
+
+  val lookup :
+       t
+    -> peer
+    -> [ `Unbanned of float  (** The peer isn't banned. *)
```

Comment:
I don't think it should be an opaque synonym, since we want to think about these things as numbers sometimes. Like maybe some action should be worth half a ban so we set its value to -0.5.

Will take the suggestion on refactoring the type.